### ISSUE TITLE
<torch/torch.h> should be <torch/script.h>

### ISSUE THREAD

**Issue by SPark9625** [pytorch/pytorch#18718]
## 📚 Documentation

On https://pytorch.org/tutorials/advanced/cpp_frontend.html and https://pytorch.org/cppdocs/frontend.html, the code that includes the pytorch library should be `#include <torch/script.h>`, not `#include <torch/torch.h>`.

**Comment by SPark9625** [comment]
I assumed that the search path was `libtorch/include`. But it seems like `libtorch/include/torch/csrc/api/include` is the right search path, am I right?

**Comment by ezyang** [comment]
@SPark9625 You want `torch/extension.h` for cpp extensions; `torch/torch.h` is valid for regular C++ use of Torch. `torch/script.h` is if you specifically want TorchScript functionality. I'm not sure about your include path question.

You do have a legitimate bug report, which is that we are recommending `torch/torch.h` when it is inappropriate. I refiled your issue in tutorials https://github.com/pytorch/tutorials/issues/473 I verified the two occurrences of this include in PyTorch repo are proper:

```
docs/cpp/source/installing.rst
47:  #include <torch/torch.h>
56:of the PyTorch C++ API, including `torch/torch.h` is the most sure-proof way of

docs/cpp/source/frontend.rst
42:  #include <torch/torch.h>
```
